### PR TITLE
Fix for ledger chunking and snapshot out-of-sync

### LINKED ISSUE #2491: Reconfiguration test: ledger files don't match up between nodes
At the end of the `reconfiguration_test_suite`, the ledger files between the primary don't seem to match up https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=23901&view=logs&j=383d248c-4494-5797-d98f-6cef5140601e&t=bbcdfa66-260b-5619-96df-9537b476c35e&l=8328:

```bash
43: Traceback (most recent call last):
43:   File "/mnt/build/2/s/tests/e2e_suite.py", line 190, in <module>
43:     run(args)
43:   File "/mnt/build/2/s/tests/e2e_suite.py", line 143, in run
43:     network.stop_all_nodes()
43:   File "/mnt/build/2/s/tests/infra/network.py", line 534, in stop_all_nodes
43:     raise Exception(
43: Exception: Ledger file on node 5326a975bc39a7ce108a29e4e58eb3135604a4cdbef4a78cfbdcd674051bbc0d does not exist on most up-to-date node 3d6694b7225c584786308d2c1295a499d057e9a15a8bd3db84f51ca1f980d26e: ledger_83-95.committed
```

We should also print clearer information about the ledger files present on each node at the end of the test run in case of error.

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves #2491 

I'd been tracking down this bug for a while, [without success](https://github.com/microsoft/CCF/pull/2526), but activating JWT auto-refresh for the test suites (see https://github.com/microsoft/CCF/pull/2600) triggers the bug much more consistently.

This is the proposed fix for it. This issue was that, while we record "snapshottable" `seqno`s as we replicate entries (on both primary and backups) based on the value of `snapshot_tx_interval`, nodes would generate snapshots on (global) commit at the `seqno` of that commit. However, commits can happen at any signature `seqno`, and don't necessarily match the "snapshottable" `seqno` which was record on the replication code path. In other words, for a `snapshot_tx_interval` of `10`, we may have recorded `seqnos` 10, 20 and 30 as "snapshottable" `seqno`s (and thus forced new ledger chunks at those) but the global commit would come at 15 and we would create a snapshot at 15, instead of 10. Then, when a new node joins from the snapshot at 15, the ledger chunks it creates would be "chunked" at different intervals than those of the primary node. 

This is very obvious in hindsight, and the snapshotter unit test now covers my original misunderstanding. I'm also very surprised that this hasn't showed up in other tests, but I'm assuming our end-to-end tests are simple enough that a service commits all committable `seqno`s?

### CODE REVIEW

File: tests/infra/logging_app.py [hunk 1]
```diff
@@ -203,11 +203,6 @@ def _verify_tx(
                             raise ValueError(
                                 f"Response with status {rep.status_code} is missing 'retry-after' header"
                             )
-                        sleep_time = 0.5
```
Comment (achamayou) [inline tests/infra/logging_app.py:206]: Why is this removed?

File: src/consensus/aft/raft.h [hunk 2]
```diff
@@ -2567,12 +2567,10 @@ namespace aft
       state->commit_idx = idx;
 
       LOG_DEBUG_FMT("Compacting...");
-      snapshotter->commit(idx);
-      if (consensus_type == ConsensusType::CFT)
-      {
-        // Snapshots are not yet supported with BFT
-        snapshotter->update(idx, replica_state == Leader);
-      }
+      // Snapshots are not yet supported with BFT
```
Comment (achamayou) [inline src/consensus/aft/raft.h:2570]: While we don't support resuming from them in BFT, it seems to me that there should be no harm in emitting them?

File: tests/infra/logging_app.py (hunk 1 above)
Comment (jumaffre) [reply tests/infra/logging_app.py:206]: There's already another sleep (line `219`) in this loop. While clients should pause for `retry-after` value returned by the historical query endpoint, in the interest of time, I suggest our e2e tests shouldn't actually wait for that long as it's likely the historical queries will be processed quickly enough?

File: src/consensus/aft/raft.h (hunk 2 above)
Comment (jumaffre) [reply src/consensus/aft/raft.h:2570]: Yes, I was wondering that too and gave it a try but this currently doesn't work as the snapshot evidence generation logic doesn't go through the frontend like BFT would expect (instead, it creates a local `kv::Tx`, records the evidence and commits it "manually"). Is creating a new endpoint for this what we want? My assumption was that all nodes should generate the same snapshot in BFT, and record identical evidences in the store when they do so.

Comment (achamayou) [reply src/consensus/aft/raft.h:2570]: If this involved creating an endpoint, let's not do it for now.

**Review by achamayou — APPROVED** [review]

**Review by achamayou — APPROVED** [review]
